force-app/main/default/classes/DMLOperations.cls

```diff
@@ -145,11 +174,45 @@
      * @param oppNames The list of Opportunity names.
      */
     public static void upsertOpportunities(String accountName, List<String> oppNames) {
+        // Try to find the account
+        Account account;
+        try {
+            account = [SELECT Id FROM Account WHERE Name = :accountName LIMIT 1];
+        } catch (QueryException e) {
+            account = null;
+        }
+    
+        // If account doesn't exist, create it
+        if (account == null) {
+            account = new Account(Name = accountName);
+            upsert account;
+        }
+    
+        // Create a list to hold the opportunities
+        List<Opportunity> oppsToUpsert = new List<Opportunity>();
+    
         // Loop through each Opportunity name
-            // Create an Opportunity with the given name
-            // Add the Opportunity to the list
+        for (String oppName : oppNames) {
+            Opportunity opp;
+            try {
+                opp = [SELECT Id FROM Opportunity WHERE Name = :oppName LIMIT 1];
```

Comment:
You should not be querying within a loop. It may work fine for single updates, but eventually, at scale, it will cause issues. Can you change this query to take multiple inputs and outside of the loop?